dts/bindings/mspi/mspi-device.yaml

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) 2024, Ambiq Micro Inc. <www.ambiq.com>
+# SPDX-License-Identifier: Apache-2.0
+
+# Common fields for MSPI devices
+
+include: [base.yaml, power.yaml]
+
+on-bus: mspi
+
+properties:
+  reg:
+    required: true
+
+  mspi-max-frequency:
+    type: int
+    required: true
+    description: |
+      Maximum clock frequency of device to configure in Hz.
+      In device tree, it is normally the target operating
+      frequency after initialization.
+
+  mspi-io-mode:
+    type: string
+    enum:
+      - "MSPI_IO_MODE_SINGLE"
+      - "MSPI_IO_MODE_DUAL"
+      - "MSPI_IO_MODE_DUAL_1_1_2"
+      - "MSPI_IO_MODE_DUAL_1_2_2"
+      - "MSPI_IO_MODE_QUAD"
+      - "MSPI_IO_MODE_QUAD_1_1_4"
+      - "MSPI_IO_MODE_QUAD_1_4_4"
+      - "MSPI_IO_MODE_OCTAL"
+      - "MSPI_IO_MODE_OCTAL_1_1_8"
+      - "MSPI_IO_MODE_OCTAL_1_8_8"
+      - "MSPI_IO_MODE_HEX"
+      - "MSPI_IO_MODE_HEX_8_8_16"
+      - "MSPI_IO_MODE_HEX_8_16_16"
+    description: |
+      MSPI I/O mode setting.
+      In device tree, it is normally the target io mode
+      after initialization.
+
+  mspi-data-rate:
+    type: string
+    enum:
+      - "MSPI_DATA_RATE_SINGLE"
+      - "MSPI_DATA_RATE_S_S_D"
+      - "MSPI_DATA_RATE_S_D_D"
+      - "MSPI_DATA_RATE_DUAL"
+    description:
+      MSPI data rate setting.
+      In device tree, it is normally the target data rate
+      after initialization.
+
+  mspi-hardware-ce-num:
+    type: int
+    description: |
+      MSPI hardware CE number.
+      MSPI controller may natively support multiple slave devices
+      on the same MSPI instance by assigning designated CE numbers.
+
+  mspi-cpp-mode:
+    type: string
+    enum:
+      - "MSPI_CPP_MODE_0"
+      - "MSPI_CPP_MODE_1"
+      - "MSPI_CPP_MODE_2"
+      - "MSPI_CPP_MODE_3"
+    description: |
+      MSPI clock polarity setting.
+      MSPI_CPP_MODE_0: CPOL=0, CPHA=0
+      MSPI_CPP_MODE_1: CPOL=0, CPHA=1
+      MSPI_CPP_MODE_2: CPOL=1, CPHA=0
+      MSPI_CPP_MODE_3: CPOL=1, CPHA=1
+
+  mspi-endian:
+    type: string
+    enum:
+      - "MSPI_LITTLE_ENDIAN"
+      - "MSPI_BIG_ENDIAN"
+    description: |
+      MSPI transfer MSB or LSB first.
+
+  mspi-ce-polarity:
+    type: string
+    enum:
+      - "MSPI_CE_ACTIVE_LOW"
+      - "MSPI_CE_ACTIVE_HIGH"
+    description: |
+      MSPI CE polarity. In most cases, it is active low.
+
+  mspi-dqs-enable:
+    type: boolean
+    description: |
+      Enable DQS mode for a device which supports it.
+      This will be checked against dqs-support and configure
+      the MSPI hardware if it supports DQS mode.
+
+  mspi-hold-ce:
+    type: boolean
+    description: |
+      In some cases, it is necessary for the master to manage
```

Comment:
"controller"

Reply:
No need to comment every single instance.
I believe the understanding is to switch to controller and target. 

Please see my comment on your other message. .